### PR TITLE
Sequence pad op

### PR DESCRIPTION
**Pull request by JiayiFeng**
Refine `sequence_padding` functor and develop `sequence_pad_op`.

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -2662,6 +2663,51 @@ def sequence_expand(x, y, ref_level=-1, name=None):
...
+            None or any positive int. When it is None, all sequences will be 
+            padded up to the length of the longest one among them; when it a 
+            certain positive value, it must be greater than the length of the 
+            longest original sequence."
+    
+    Returns:
+        Variable: The padded sequence batch. All sequences has the same length.
+    
+    Examples:
+        .. code-block:: python
```
Comment (panyx0718) [inline python/paddle/fluid/layers/nn.py:2681]: better to have a formula or example of tensor before/after pad.

**Review by panyx0718 — DISMISSED** [review]
LG API

**Review by reyoung — DISMISSED** [review]

File: python/paddle/fluid/layers/nn.py (hunk 1 above)
Comment (JiayiFeng) [reply python/paddle/fluid/layers/nn.py:2681]: We do have demos. It is exposed to Python API reference by `${comment}`

https://github.com/PaddlePaddle/Paddle/pull/12824/files#diff-4012ff792c98c6e3a2d5504329ae32caR117

**Review by reyoung — DISMISSED** [review]

**Review by panyx0718 — DISMISSED** [review]
LG API

**Review by panyx0718 — APPROVED** [review]

**Review by reyoung — APPROVED** [review]

**Review by kuke — APPROVED** [review]
LGTM

**Pull request merged**